### ISSUE TITLE
Support additional group/decimal separators

### ISSUE THREAD

**Issue by btrautmann** [onepub-dev/money.dart#79]
![Screenshot 2023-12-19 at 06 15 15](https://github.com/onepub-dev/money.dart/assets/8343465/5cbc3f12-8eb8-4897-8e02-2625a3ca0b49)

I'm working with an API that supports a bunch of different currency formats and I'm bumping up against the restriction of only allowing the use of `,` and `.` characters. The API requires me to support spaces for group separators and `/` and `-` as decimal separators. It would be awesome if this was supported natively by this package.

In the meantime, I'll likely have to substitute the characters for substituted ones and then replace them after formatting.

**Comment by btrautmann** [comment]
Here's where I've landed:

```dart
if (currencyFormat case Some<CurrencyFormat>(:final some)) {
  // Example patterns are provided by the API without the currency
  // symbol.
  // - Replace all digits with `#`
  // - Use 0s to denote post-decimal precision
  // - Add symbol back in at the beginning or end of the pattern, as needed.
  String createPattern() {
    // e.g 123,456.789
    final sourceFormat = some.exampleFormat;
    final groupSeparator = some.groupSeparator;
    final decimalSeparator = some.decimalSeparator;
    // e.g ###,###.###
    final hashes = sourceFormat.replaceAll(RegExp(r'\d'), '#');
    final decimalIndex = hashes.indexOf(some.decimalSeparator);
    final zeros = List.generate(some.decimalDigits, (i) => '0').join();
    // e.g ###,###.000
    final pattern = '${hashes.substring(0, decimalIndex)}${some.decimalSeparator}$zeros' //
        // Money2 doesn't support special decimal separators, so use standard
        // ones here and then add them back post-format
        .replaceAll(decimalSeparator, '.')
        // Money2 doesn't support special group separators, so use standard
        // ones here and then add them back post-format
        .replaceAll(groupSeparator, ',');
    // e.g S###,###.000
    final withSymbol = some.shouldDisplaySymbol
        ? some.isSymbolFirst
            ? 'S$pattern'
            : '${pattern}S'
        : pattern;
    return withSymbol;
  }

  String symbol() => some.shouldDisplaySymbol ? some.currencySymbol : '';

  final pattern = createPattern();
  final currency = Currency.create(
    some.isoCode,
    some.decimalDigits,
    symbol: symbol(),
    // Always pass false here because we use our own separators
    // until the very end.
    invertSeparators: false,
    pattern: pattern,
  );
  final minorUnits = absMinorUnitsForDecimalDigits(some.decimalDigits);
  final money = Money.fromIntWithCurrency(minorUnits, currency, scale: some.decimalDigits);
  final formatted = formatWithSign(money);
  return formatted
      // Add back custom group separators
      .replaceAll(',', some.groupSeparator) //
      // Add back custom decimal separators
      .replaceAll('.', some.decimalSeparator);
```

And the following test passes:

```dart
test('decimal separator as slash', () {
  const euroCurrency = CurrencyFormat(
    isoCode: 'EUR',
    decimalDigits: 2,
    decimalSeparator: '/',
    groupSeparator: ' ',
    currencySymbol: '€',
    isSymbolFirst: false,
    shouldDisplaySymbol: true,
    exampleFormat: '123 456/78',
  );

  const amount = 1234567890;
  final formatted = amount.formatted(const Some(euroCurrency));
  expect(formatted, '1 234 567/89€');
});
```

**Comment by bsutton** [comment]
I've just released 5.0.0-alpha1 which now supports custom group and decimal separators.

The following unit test now succeeds.

Any feed back would be appreciated.


``` dart
   test('decimal separator as slash', () {
      final euroCurrency = Currency.create('EUR', 2,
          decimalSeparator: '/',
          groupSeparator: ' ',
          symbol: '€',
          pattern: '###,###.##S');

      final amount =
          Money.fromIntWithCurrency(1234567890, euroCurrency, decimalDigits: 3);
      final formatted = amount.toString();
      expect(formatted, '1 234 567/89€');
    });
```

**Comment by btrautmann** [comment]
That's awesome @bsutton. I'm in the middle of a huge refactor but once I'm out of the woods with that I will give this a shot!
